services/galley/src/Galley/API/MLS/Message.hs

```diff
@@ -893,9 +901,24 @@ processInternalCommit qusr senderClient con lConvOrSub epoch action senderRef co
                     senderRef'
             -- remote clients cannot send the first commit
             (False, _, _, _) -> throwS @'MLSStaleMessage
-            (_, _, _, SubConv _ _) -> pure ()
+            (True, _, [], SubConv parentConv _) -> do
+              creatorClient <- noteS @'MLSMissingSenderClient senderClient
+              unless (isClientMember (mkClientIdentity qusr creatorClient) (mcMembers parentConv)) $
+                throwS @'MLSSubConvClientNotInParent
+              creatorRef <-
+                maybe
+                  (pure senderRef)
+                  ( note (mlsProtocolError "Could not compute key package ref")
+                      . kpRef'
+                      . upLeaf
+                  )
+                  $ cPath commit
+              addMLSClients
+                (cnvmlsGroupId mlsMeta)
+                qusr
+                (Set.singleton (creatorClient, creatorRef))
```

Comment:
We've had this pattern of validation and addition before, so perhaps it'd be good to extract it into a function, maybe in a separate PR.

Reply:
The validation part I could factor out I think: https://github.com/wireapp/wire-server/pull/2945